details.js

```diff
@@ -0,0 +1,34 @@
+const params = new URLSearchParams(window.location.search);
+const countryName = params.get("country");
+
+let country = {};
+
+const requestCountry = async () => {
+    try {
+        const response = await fetch(`https://restcountries.com/v3.1/name/${countryName}?fullText=true`);
+        if (!response.ok) {
+            throw new Error('Network response was not ok.');
+        }
+        const data = await response.json();
+        country = data[0];
+        insertValues(country);
+    } catch (error) {
+        console.error('Error fetching country data:', error);
+    }
+}
+
+requestCountry();
+
+const insertValues = (country) => {
```

Comment:
consider changing the function's name to be more understandable.
consider: `renderCountryDetails`